nucleus/core/kernel/src/main/java/org/glassfish/kernel/event/EventsImpl.java

```diff
@@ -82,7 +82,7 @@ public void send(final Event event, boolean asynchronously) {
                 logger.log(Level.SEVERE, KernelLoggerInfo.exceptionSendEvent, ex);
             }
             if (m!=null) {
-                RestrictTo fooBar = m.getParameterTypes()[0].getAnnotation(RestrictTo.class);
```

Comment:
`getParameterTypes()` method returns types of the parameters. In our case this will be `Event.class`. This class itself does not contain the `RestrictTo` annotation, thus its `getAnnotation(RestrictTo.class)` method wili return `null`.

On the other side, `getParameters()` method returns an array of the `java.lang.reflect.Parameter` and `Parameter.getAnnotation(RestrictTo.class)` will return an instance of the `RestrictTo` annotation if the method parameter contains this annotation.